artcommon/artcommonlib/konflux/konflux_db.py

```diff
@@ -234,10 +237,40 @@ async def get_latest_build(self, name: str, group: str, outcome: KonfluxBuildOut
                 continue
 
         # If we got here, no builds have been found in the whole 36 months period
+        if strict:
+            raise ValueError(f"Build record for {name} not found.")
         self.logger.warning('No builds found for %s in %s with status %s in assembly %s and target %s',
                             name, group, outcome.value, assembly, el_target)
         return None
 
+    async def get_build_record_by_nvr(self, nvr: str, outcome: KonfluxBuildOutcome = KonfluxBuildOutcome.SUCCESS, strict: bool = True) -> typing.Optional[KonfluxRecord]:
+        """ Get a build record by NVR.
+
+        :param nvr: The NVR of the build.
+        :param outcome: The outcome of the build.
+        :param strict: If True, raise an exception if the build record is not found.
+        :return: The build record; None if the build record is not found.
+        :raise: IOError if the build record is not found and strict is True.
+        """
+        where = {"nvr": nvr, "outcome": str(outcome)}
+        started_before = datetime.now(tz=timezone.utc)
+        for window in range(12):
```

Comment:
Since `start_time` is something we always have to provide `search_builds_by_fields()` with, I think it could useful to embed this window search into that function. the `start_search` param could be made optional, and if None we would go through this loop. this way, you can get rid of the new `get_build_record_by_nvr()` function and just call `search_builds_by_fields()` appropriately

Reply:
I feel it might be reasonable to have `search_builds_by_fields` only search for once while adding another wrapper function to do windowed searches. It will also keep the function simpler. 
I added `search_builds_by_fields_with_window`